alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -109,6 +113,19 @@ private void updateCompensationStatus(TxEvent event) {
     markSagaEnded(event);
   }
 
+  private void abortTimeoutEvent() {
+    eventRepository.findFirstTimeoutEventByIdGreaterThan(nextTimeoutEventId)
```

Comment:
checking timeout cannot be done by scanning through event ids, because some to-be-expired event will be skipped.